File: 최경열/이코테/정렬/퀵정렬/quick2.py
arr=[5,7,9,0,3,1,6,2,4,8]

def quick_sort(arr):
    if len(arr)<=1:
        return arr
    pivot = arr[0]
    tail =arr[1:0]
    left_side =[x for in tail if x<=pivot]
    right_side=[x for in tail if x>=pivot]
    
    return quick_sort(left_side) + [pivot] + quick_sort(right_side)

print(arr)

File: 최경열/이코테/정렬/퀵정렬/quick2.js
//es5
var arr = [5, 7, 9, 0, 3, 1, 6, 2, 4, 8];
function quickSort(arr) {
  if (arr.length <= 1) {
    return arr;
  }
  var pivot = arr[0];
  var tail = arr.slice(1);
  var leftSide = tail.filter((x) => x <= pivot);
  var rightSide = tail.filter((x) => x >= pivot);

  return quickSort(leftSide).concat(pivot, quickSort(rightSide));
}
console.log(quickSort(arr));

//es6
const arr = [5, 7, 9, 0, 3, 1, 6, 2, 4, 8];
const quickSort = (arr) => {
  if (arr.length <= 1) {
    return arr;
  }

  const pivot = arr[0];
  const tail = arr.slice(1);
  const leftSide = tail.filter((x) => x <= pivot);
  const rightSide = tail.filter((x) => x > pivot);

  return [...quickSort(leftSide), pivot, ...quickSort(rightSide)];
};

console.log(quickSort(arr));
